src/DataBus/AzureDataBusPersistence.cs

```diff
@@ -14,11 +20,63 @@ protected override void Setup(FeatureConfigurationContext context)
         {
             var dataBusSettings = context.Settings.GetOrDefault<DataBusSettings>() ?? new DataBusSettings();
 
-            var cloudBlobClient = CloudStorageAccount.Parse(dataBusSettings.ConnectionString).CreateCloudBlobClient();
+            ThrowIfConnectionStringAndTokenProviderSpecified(dataBusSettings);
 
-            var dataBus = new BlobStorageDataBus(cloudBlobClient.GetContainerReference(dataBusSettings.Container), dataBusSettings, new AsyncTimer());
+            var container = CreateCloudBlobContainer(dataBusSettings);
+
+            var dataBus = new BlobStorageDataBus(container, dataBusSettings, new AsyncTimer());
 
             context.Container.ConfigureComponent(b => dataBus, DependencyLifecycle.SingleInstance);
         }
+
+        static void ThrowIfConnectionStringAndTokenProviderSpecified(DataBusSettings dataBusSettings)
+        {
+            if (dataBusSettings.RenewalTimeBeforeTokenExpires != null && dataBusSettings.UserProvidedConnectionString)
+            {
+                throw new Exception("More than one method to connect to the storage account was supplied (using connection string and token provider). Use one method only.");
+            }
+        }
+
+        static CloudBlobContainer CreateCloudBlobContainer(DataBusSettings dataBusSettings)
+        {
+            CloudBlobContainer container;
+
+            // Attempt managed identity identity first
+            if (!string.IsNullOrWhiteSpace(dataBusSettings.StorageAccountName))
+            {
+                var azureServiceTokenProvider = new AzureServiceTokenProvider();
+                var state = (azureServiceTokenProvider, dataBusSettings.RenewalTimeBeforeTokenExpires);
+                var tokenAndFrequency = TokenRenewerAsync(state, CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
Should this not be an `async` method?

Reply:
Won't help. It's called from the Setup method of `Feature` that is not returning `Task`.

Reply:
:+1: